modules/apps/portal-odata/portal-odata-impl/src/main/java/com/liferay/portal/odata/internal/filter/expression/ExpressionVisitorImpl.java

```diff
@@ -316,6 +316,9 @@ else if (binaryOperatorKind == BinaryOperatorKind.NE) {
 		else if (binaryOperatorKind == BinaryOperatorKind.OR) {
 			return Optional.of(BinaryExpression.Operation.OR);
 		}
+		else if (binaryOperatorKind == BinaryOperatorKind.SUB) {
```

Comment:
We should delete the Optional for the code, per Brian request, this method should return a BinaryExpression.Operation or an Optional.

Plese, cherry pick https://github.com/liferay-tango/liferay-portal/pull/3152/commits/e3172a80ad7bcd3e43304eb13e858303d6e2c12d